In [1]:
import pandas as pd 

In [3]:
# BOY MAP Data 
boy = pd.read_csv("Data/MAP_BOY_2020-21.csv") 

# MOY MAP Data 
moy = pd.read_csv("Data/MAP_MOY_2020-21.csv") 

# student demographic data
demographics = pd.read_csv("Data/Student_Demographics_2020-21.csv")

# EDA 

## BOY MAP DATA

In [4]:
boy.head()

Fake_Student_ID        TermName                               DistrictName  \
0          2115893  Fall 2020-2021  Dallas Independent School District (DISD)   
1          2115893  Fall 2020-2021  Dallas Independent School District (DISD)   
2          5090051  Fall 2020-2021  Dallas Independent School District (DISD)   
3          5090051  Fall 2020-2021  Dallas Independent School District (DISD)   
4          8221829  Fall 2020-2021  Dallas Independent School District (DISD)   

                            SchoolName StudentDateOfBirth  Grade  \
0    Harry C Withers Elementary School          9/12/2012      2   
1    Harry C Withers Elementary School          9/12/2012      2   
2  Louise Wolff Kahn Elementary School          5/14/2014      1   
3  Louise Wolff Kahn Elementary School          5/14/2014      1   
4   Stephen C Foster Elementary School          10/5/2012      2   

         Subject             Course           TestType  \
0    Mathematics          Math K-12  Survey With Goals   
1  Language Arts            Reading  Survey With Goals   
2    Mathematics          Math K-12  Survey With Goals   
3  Language Arts  Reading (Spanish)  Survey With Goals   
4    Mathematics          Math K-12  Survey With Goals   

                              TestName TestStartDate TestStartTime  \
0             Growth: Math 2-5 TX 2012     9/29/2020       9:04:36   
1          Growth: Reading 2-5 TX 2017     9/22/2020       8:45:17   
2             Growth: Math K-2 TX 2012     9/21/2020       9:19:13   
3  Growth: Spanish Reading K-2 TX 2017     9/21/2020      10:57:10   
4             Growth: Math 2-5 TX 2012     9/21/2020       9:07:44   

   TestDurationMinutes  TestRITScore  TestStandardError  TestPercentile  \
0                   38           187                3.0              82   
1                   94           197                3.5              95   
2                   37           153                3.2              29   
3                   18           132                3.5               9   
4                   41           142                3.3               1   

  AchievementQuintile  PercentCorrect  
0                High              50  
1                High              60  
2               LoAvg              49  
3                 Low              37  
4                 Low              30

In [5]:
#columns in dataframe
boy.columns

Index(['Fake_Student_ID', 'TermName', 'DistrictName', 'SchoolName',
       'StudentDateOfBirth', 'Grade', 'Subject', 'Course', 'TestType',
       'TestName', 'TestStartDate', 'TestStartTime', 'TestDurationMinutes',
       'TestRITScore', 'TestStandardError', 'TestPercentile',
       'AchievementQuintile', 'PercentCorrect'],
      dtype='object')

In [6]:
#total number of grades represented 
boy['Grade'].unique()

array([2, 1])

Data is from students in 1st & 2nd grade.

In [7]:
#Districts 
boy['DistrictName'].unique()

array(['Dallas Independent School District (DISD)'], dtype=object)

In [12]:
#schools represented  
print(f"Number of Schools {len(boy['SchoolName'].unique())}")
print(boy['SchoolName'].unique())

Number of Schools 149
['Harry C Withers Elementary School' 'Louise Wolff Kahn Elementary School'
 'Stephen C Foster Elementary School' 'Ida B. Wells Academy'
 'KB Polk Center for Academically Talented & Gifted'
 'William Lipscomb Elementary School' 'Chapel Hill Preparatory'
 'Mockingbird Elementary School' 'Obadiah Knight Elementary School'
 'Maple Lawn Elementary School'
 'Celestino Mauricio Soto Jr Elementary School'
 'S S Conner Elementary School' 'L L Hotchkiss Elementary School'
 'Edna Rowe Elementary School' 'Central Elementary School'
 'Charles Rice Learning Center' 'Anson Jones Elementary School'
 'James Bowie Elementary School' 'Arturo Salazar Elementary School'
 'Casa View Elementary School' 'Winnetka Elementary School'
 'Arthur Kramer Elementary School'
 'Thomas L Marsalis Elementary STEAM Academy'
 'Lee A McShan Jr Elementary School' 'Edwin J Kiest Elementary School'
 'Larry G Smith Elementary School' 'George Peabody Elementary School'
 'W W Bushman Elementary School' 'Dani

All data is from DISD School, there are 149 schools represented in this data set. 

In [16]:
#descriptive statistics for quantitative columns
boy[['TestDurationMinutes', 'TestRITScore', 'TestStandardError', 'TestPercentile','PercentCorrect']].describe()

TestDurationMinutes  TestRITScore  TestStandardError  TestPercentile  \
count         38648.000000  38648.000000       38648.000000    38648.000000   
mean             41.486416    162.801671           3.281049       45.061374   
std              27.827051     18.215424           0.289124       31.295397   
min               6.000000    102.000000           2.900000        1.000000   
25%              23.000000    150.000000           3.100000       17.000000   
50%              34.000000    162.000000           3.200000       41.000000   
75%              51.000000    174.000000           3.400000       72.000000   
max             339.000000    259.000000          13.400000       99.000000   

       PercentCorrect  
count    38648.000000  
mean        47.387032  
std          8.821543  
min          8.000000  
25%         44.000000  
50%         49.000000  
75%         53.000000  
max        100.000000